.github/workflows/technology-prepare-release.yml

```diff
@@ -0,0 +1,78 @@
+name: Prepare Release
+
+on:
+  workflow_call:
+    inputs:
+      branch:
+        description: the starting branch for the release. If left empty "main" will be used, making this a proper release, otherwise it will be a bugfix
+        required: false
+        type: string
+        default: main
+      version:
+        description: the version to be released. if "branch" is "main" this version number should end with 0 (e.g. 0.1.0), otherwise it must end with a number greater than 0
+        required: true
+        type: string
+
+jobs:
+  Prepare-Release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+
+      - name: set release type
+        shell: bash
+        id: vars
+        run: |
+          type=release
+          if [[ ${{ inputs.branch }} != "main" ]] 
+          then
+            type=bugfix
+          fi
+          echo "type=$type" >> $GITHUB_OUTPUT
+
+      - name: create ${{ steps.vars.outputs.type }}/${{ inputs.version }} branch
+        shell: bash
+        run: |
+          git config user.name "eclipse-edc-bot"
+          git config user.email "edc-bot@eclipse.org"
+          
+          git checkout -b ${{ steps.vars.outputs.type }}/${{ inputs.version }}
+          git push -f origin ${{ steps.vars.outputs.type }}/${{ inputs.version }}
+
+      - shell: bash
+        run: |
+          # updates the project version
+          sed -i 's#^version=.*#version=${{ inputs.version }}#g' $(find . -name "gradle.properties")
+
+          # updates the eventual core library version in the version catalog
+          sed -i 's#^edc = .*#edc = "${{ inputs.version }}"#g' gradle/libs.versions.toml
+
```

Comment:
Technically, we'd also have to update the version in the `DEPENDENCIES` file, otherwise we'd be releasing artefacts with wrong dependencies listed.

Since we have to do that, how about simply parsing the "old" version from `gradle.properties` like so:
```bash
existingVersion=$(grep "version" gradle.properties  | awk -F= '{print $2}')
```

and then all files could be `sed`-ed in one go:
```bash
grep -rlz "$existingVersion" . --exclude=\*.{sh,bin} | xargs sed -i "s/$existingVersion/${{ inputs.version }}/g"
```

Reply:
the nice thing about this new process is that the `preparation` phase will create a PR that will need to be merged manually, so in that one all the checks should run, dependencies one included, at that point the committer will need to fix everything (and eventually, for a bugfix, cherry-picking the fix commits) before merging it and trigger the release